rsvim_core/src/ui/widget/command_line.rs

```diff
@@ -101,46 +107,67 @@ impl CommandLine {
     let cmdline_content_shape =
       IRect::new((shape.min().x + 1, shape.min().y), shape.max().into());
 
-    let (viewport, cursor_viewport) = {
+    let (input_viewport, input_cursor_viewport, message_viewport) = {
       let cmdline_content_actual_shape =
         geo_rect_as!(cmdline_content_shape, u16);
       let text_contents = text_contents.upgrade().unwrap();
       let text_contents = lock!(text_contents);
-      let viewport = Viewport::view(
+      let input_viewport = Viewport::view(
         &options,
         text_contents.command_line_content(),
         &cmdline_content_actual_shape,
         0,
         0,
       );
-      let cursor_viewport = CursorViewport::from_top_left(
-        &viewport,
+      let input_cursor_viewport = CursorViewport::from_top_left(
+        &input_viewport,
         text_contents.command_line_content(),
       );
-      (viewport, cursor_viewport)
+      let message_viewport = Viewport::view(
+        &options,
+        text_contents.command_line_message(),
+        &cmdline_content_actual_shape,
+        0,
+        0,
+      );
+      (input_viewport, input_cursor_viewport, message_viewport)
     };
-    let viewport = Viewport::to_arc(viewport);
-    let cursor_viewport = CursorViewport::to_arc(cursor_viewport);
+    let input_viewport = Viewport::to_arc(input_viewport);
+    let input_cursor_viewport = CursorViewport::to_arc(input_cursor_viewport);
+    let message_viewport = Viewport::to_arc(message_viewport);
 
     let cmdline_content = CommandLineContent::new(
       cmdline_content_shape,
       text_contents.clone(),
-      Arc::downgrade(&viewport),
+      Arc::downgrade(&input_viewport),
     );
     let cmdline_content_id = cmdline_content.id();
-    let cmdline_content_node =
+    let mut cmdline_content_node =
       CommandLineNode::CommandLineContent(cmdline_content);
+    cmdline_content_node.set_visible(false);
```

Comment:
In [line-104](https://github.com/rsvim/rsvim/pull/559/files#diff-a8498a0344aa604e2c844243cb3deb8bd1b2e5da1e065f5e7dc4e4a94e77a8b9R104), I think `cmdline_indicator_node` should also set `set_visible(false)`. The "indicator" and "content" should always show/hide together.